duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -490,19 +676,20 @@ namespace caf {
   {
     fMetaTree->Fill();
 
-    if(fOutFile){
-      fOutFile->cd();
-      fTree->Write();
-      fMetaTree->Write();
-      fOutFile->Close();
-    }
+    fOutFile->cd();
+    fTree->Write();
+    fMetaTree->Write();
+    fGENIETree->Write();
```

Comment:
The TFileService could be used to manage the root objects and avoid doing manual writes